### PR TITLE
chore(loader): convert enzyme to RTL - FE-6607

### PR DESCRIPTION
**Pull request by DipperTheDan**
### Proposed behaviour

- Refactor unit tests to RTL
- Add Playwright test to assert the correct CSS values are applied to the `gradient` version of Loader.

### Current behaviour

Tests currently use Enzyme.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context

I have added comments where tests are required for coverage. 

### Testing instructions

N/A

### CODE REVIEW

File: src/components/loader/loader.test.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
```
Comment (edleeks87) [inline src/components/loader/loader.test.tsx:3]: comment: I don't think we need to import this in either file, if we do it should be done in the jest config (I think it is already but could be wrong)

File: src/components/loader/loader.test.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import useMediaQuery from "../../hooks/useMediaQuery";
+
+jest.mock("../../hooks/useMediaQuery", () => ({
+  __esModule: true,
+  default: jest.fn().mockReturnValue(false),
+}));
+
+testStyledSystemMargin((props) => <Loader {...props} />);
+
+test("if the user disallows animations or their preference cannot be determined, alternative loading text is rendered", () => {
```
Comment (edleeks87) [inline src/components/loader/loader.test.tsx:15]:
```suggestion
test("when the user disallows animations or their preference cannot be determined, alternative loading text is rendered", () => {
```

non-blocking

File: src/components/loader/loader.test.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  render(<Loader />);
+
+  expect(screen.getByText("Loading")).toBeInTheDocument();
+});
+
+describe("if the user allows animations", () => {
```
Comment (edleeks87) [inline src/components/loader/loader.test.tsx:21]:
```suggestion
describe("when the user allows animations", () => {
```

non-blocking

File: src/components/loader/loader-square.test.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  });
+});
+
+// These styling tests required for coverage
+test("when inside button and `isActive` prop is false, the expected background colour is applied", () => {
+  render(<Loader isInsideButton isActive={false} />);
+
+  const loaderSquares = screen.getAllByTestId("loader-square");
+
+  expect(loaderSquares[0]).toHaveStyle({
```
Comment (edleeks87) [inline src/components/loader/loader-square.test.tsx:88]: comment: I think we need to use toHaveStyleRule here as the jest DOM matcher returns false positives when testing token values

**Review by edleeks87 — DISMISSED** [review]

File: src/components/loader/loader.test.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    render(<Loader />);
+
+    expect(screen.getByLabelText("Loading")).toBeInTheDocument();
+  });
+
+  test("when custom `aria-label` is passed, set accessible name to its value", () => {
+    render(<Loader aria-label="Still loading" />);
+
+    expect(screen.getByLabelText("Still loading")).toBeInTheDocument();
+  });
```
Comment (Parsium) [inline src/components/loader/loader.test.tsx:46]: **suggestion(non-blocking)**: what do you think of using the `toHaveAccessibleName` assertion, so the error messaging aligns more closely to each test's objective?:

```suggestion
    expect(screen.getByRole("progressbar")).toHaveAccessibleName("Loading");
  });

  test("when custom `aria-label` is passed, set accessible name to its value", () => {
    render(<Loader aria-label="Still loading" />);

    expect(screen.getByRole("progressbar")).toHaveAccessibleName("Still loading");
  });
```

Comment (DipperTheDan) [reply src/components/loader/loader.test.tsx:46]: Yeah, I'm happy to do that. I'll get on it now ✈️

**Review by Parsium — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by stephenogorman — APPROVED** [review]

**Pull request merged**